keanu-project/src/test/java/io/improbable/keanu/algorithms/NetworkSamplesTest.java

```diff
@@ -127,4 +144,59 @@ public void canBeConstructedFromListOfNetworkSample() {
         assertEquals(logOfMasterPBySample.get(0), networkSamples.getLogOfMasterP(0));
         assertEquals(logOfMasterPBySample.get(1), networkSamples.getLogOfMasterP(1));
     }
+
+    @Test
+    public void getReturnsDoubleVertexSamplesIfVariableIsDoubleVertex() {
+        DoubleVertex vertex = new GaussianVertex(0., 1.);
+        Map<VariableReference, DoubleTensor> vertexValsFirstSample = ImmutableMap.of(
+            vertex.getId(), DoubleTensor.scalar(0.)
+        );
+        NetworkSamples networkSamples = NetworkSamples.from(ImmutableList.of(
+            new NetworkSample(vertexValsFirstSample, 9.4)
+        ));
+        assertThat(vertex, instanceOf(DoubleVertex.class));
+        assertThat(networkSamples.get(vertex), instanceOf(DoubleVertexSamples.class));
+    }
+
+    @Test
+    public void getReturnsIntegerSamplesIfVariableIsIntegerVertex() {
+        IntegerVertex vertex = new UniformIntVertex(0, 1);
+        Map<VariableReference, IntegerTensor> vertexValsFirstSample = ImmutableMap.of(
+            vertex.getId(), IntegerTensor.scalar(0)
+        );
+        NetworkSamples networkSamples = NetworkSamples.from(ImmutableList.of(
+            new NetworkSample(vertexValsFirstSample, 9.4)
+        ));
+        assertThat(vertex, instanceOf(IntegerVertex.class));
+        assertThat(networkSamples.get(vertex), instanceOf(IntegerVertexSamples.class));
+    }
+
+    @Test
+    public void getReturnsBooleanSamplesIfVariableIsBooleanVertex() {
+        BooleanVertex vertex = new BernoulliVertex(0.5);
+        Map<VariableReference, BooleanTensor> vertexValsFirstSample = ImmutableMap.of(
+            vertex.getId(), BooleanTensor.scalar(true)
+        );
+        NetworkSamples networkSamples = NetworkSamples.from(ImmutableList.of(
+            new NetworkSample(vertexValsFirstSample, 9.4)
+        ));
+        assertThat(vertex, instanceOf(BooleanVertex.class));
+        assertThat(networkSamples.get(vertex), instanceOf(BooleanVertexSamples.class));
+    }
+
+    @Test
+    public void getReturnsSamplesIfVariableIsGenericVertex() {
+        GenericVertex<Integer> vertex = new ConstantGenericVertex(0);
+        Map<VariableReference, GenericTensor<Integer>> vertexValsFirstSample = ImmutableMap.of(
+            vertex.getId(), GenericTensor.scalar(0)
+        );
+        NetworkSamples networkSamples = NetworkSamples.from(ImmutableList.of(
+            new NetworkSample(vertexValsFirstSample, 9.4)
+        ));
+        assertThat(vertex, instanceOf(GenericVertex.class));
+        assertThat(networkSamples.get(vertex), not(instanceOf(DoubleVertexSamples.class)));
+        assertThat(networkSamples.get(vertex), not(instanceOf(IntegerVertexSamples.class)));
+        assertThat(networkSamples.get(vertex), not(instanceOf(BooleanVertexSamples.class)));
+        assertThat(networkSamples.get(vertex), instanceOf(Samples.class));
```

Comment:
you could replace these 4 lines with:

        assertThat(networkSamples.get(vertex).getClass(), equalTo(Samples.class));